tsdb/head.go

```diff
@@ -2516,12 +2517,14 @@ func (s *memSeries) cleanupAppendIDsBelow(bound uint64) {
 // computeChunkEndTime estimates the end timestamp based the beginning of a
 // chunk, its current timestamp and the upper bound up to which we insert data.
 // It assumes that the time range is 1/4 full.
+// Assuming that the samples will keep arriving at the same rate, it will make the
+// remaining n chunks within this chunk range (before max) equally sized.
 func computeChunkEndTime(start, cur, max int64) int64 {
-	a := (max - start) / ((cur - start + 1) * 4)
-	if a == 0 {
+	n := (max - start) / ((cur - start + 1) * 4)
+	if n <= 1 {
```

Comment:
if `n == 1` then we don't need to calculate `start + (max-start)/n` as that is just `start + (max-start)/1 = start + max - start = max`, plus IMO it's even more readable, as it says _if we're going to fill at most 1 chunk, then keep using the same `max`_